api/tests/app/test_camera_metrics.py

```diff
@@ -0,0 +1,903 @@
+                          '2020-09-23 2020-09-29']
+            }),
+            ("face-mask-detections", {
+                'no_face': [88, 85, 106, 85], 'face_with_mask': [310, 519, 445, 519],
+                'face_without_mask': [150, 171, 180, 171],
+                'weeks': ['2020-09-02 2020-09-08', '2020-09-09 2020-09-15', '2020-09-16 2020-09-22',
+                          '2020-09-23 2020-09-29']
+            })
+        ]
+    )
+    @freeze_time("2020-09-30")
+    def test_get_a_weekly_report_properly_weeks_value_and_dates(self, config_rollback_create_cameras,
+                                                                reports_simulation, metric, expected):
+        """
+        Here we mock datetime.date.today() to a more convenient date set in @freeze_time("2012-01-01")
+        In addition, query string weeks is given, but also from_date and to_date. So dates should be ignored.
+        """
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+        camera_id = camera["id"]
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?cameras={camera_id}&weeks=4&from_date=2020-09-21&to_date=2020"
+            f"-09-27")
+
+        assert response.status_code == 200
+        assert response.json() == expected
+
+    @pytest.mark.parametrize(
+        "metric,expected",
+        [
+            ("social-distancing", {'detail': "Camera with id 'BAD_ID' does not exist"}),
+            ("face-mask-detections", {'detail': "Camera with id 'BAD_ID' does not exist"})
+        ]
+    )
+    def test_try_get_a_weekly_report_bad_id(self, config_rollback_create_cameras, reports_simulation, metric, expected):
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+        camera_id = 'BAD_ID'
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?cameras={camera_id}&from_date=2020-09-20&to_date=2020-09-23")
+
+        assert response.status_code == 404
+        assert response.json() == expected
+
+    @pytest.mark.parametrize(
+        "metric,expected",
+        [
+            ("social-distancing", {
+                'detected_objects': [], 'no_infringement': [], 'low_infringement': [], 'high_infringement': [],
+                'critical_infringement': [], 'weeks': []
+            }),
+            ("face-mask-detections", {
+                'no_face': [], 'face_with_mask': [], 'face_without_mask': [], 'weeks': []
+            })
+        ]
+    )
+    def test_get_a_weekly_report_no_query_string(self, config_rollback_create_cameras, reports_simulation,
+                                                 metric, expected):
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly")
+
+        # TODO: IF THIS BEHAVIOUR IS RIGHT
```

Comment:
I think it should be filled with zeroes,  so this could be a bug, we should ask the frontend guys